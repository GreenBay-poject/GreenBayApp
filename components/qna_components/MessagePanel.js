import React, { useEffect, useState } from 'react';
import { ActivityIndicator, Button, Dimensions, PixelRatio } from 'react-native';
import { SafeAreaView, ScrollView, StyleSheet, Text, View } from 'react-native';
import MapView, { Marker } from 'react-native-maps';
import Bubble from '../../atoms/Bubble';
import PostCard from '../../atoms/post';
import { requestPublicNotes } from '../../server/noterequestgenerator';

export default function MessagePanel({ set_ministry_selected, messagetoshow }) {

    const [qna_elem, set_qna_elem] = useState([])


    // useEffect(() => {
    //     var qna_elem1 = []
    //     var i = 0
    //     messagetoshow.forEach(element => {
    //         i += 1
    //         qna_elem1.push(
    //             <Bubble key={i} by={element['uname']} ans={element['answer']} title={element['title']} question={element['question']} date={element['dateposted']} />

    //         )
    //     });
    //     set_qna_elem(qna_elem1)

    // }, [messagetoshow]);



    return (
        <ScrollView style={{ height: '80%' }} >

            {qna_elem}


        </ScrollView>

    );
}

const styles = StyleSheet.create({

});