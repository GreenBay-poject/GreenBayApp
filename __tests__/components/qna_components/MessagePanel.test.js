import React from 'react';
import renderer from 'react-test-renderer';
import MessagePanel from '../../../components/qna_components/MessagePanel';

describe('<MessagePanel />', () => {
    it('Sample MessagePanel', () => {
        const tree = renderer.create(<MessagePanel set_ministry_selected={null} messagetoshow={"Testing"} />).toJSON();
        console.log(tree)
        expect(tree).toMatchSnapshot();
        expect(tree.type).toBe('RCTScrollView');
        expect(tree.children[0].type).toBe("View");
        // expect(tree.children[1].type).toBe("TextInput");
        // expect(tree.children[2].type).toBe("TextInput");
        // expect(tree.children[3].type).toBe("View");
        // expect(tree.children[4].type).toBe("Text");


    });
});