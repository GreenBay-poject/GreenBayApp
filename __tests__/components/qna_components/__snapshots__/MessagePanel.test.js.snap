// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagePanel /> Sample MessagePanel 1`] = `
<RCTScrollView
  style={
    Object {
      "height": "80%",
    }
  }
>
  <View />
</RCTScrollView>
`;
